rules/Naming/ExpectedNameResolver/MatchParamTypeExpectedNameResolver.php

```diff
@@ -28,9 +30,9 @@ public function resolve(Param $param): ?string
 
         $staticType = $this->staticTypeMapper->mapPhpParserNodePHPStanType($param->type);
 
-        // include nullable too
+        // include nullabgitle too
         // skip date time + date time interface, as should be kept
-        if ($this->isDateTimeType($staticType)) {
+        if ($this->nodeTypeResolver->isObjectType($param->type, new ObjectType('DateTimeInterface'))) {
```

Comment:
@TomasVotruba using `isObjectType()` handle it correctly as `DateTimeImmutable` is child of `DateTimeInterface` 👍